### PR TITLE
feat: add macos arm64 build

### PR DESCRIPTION
**Pull request by kalidax**
Add cross compilation for supporting macOS ARM64
see dry-run (no upload to release and s3) here : https://github.com/axelarnetwork/tofnd/actions/runs/2568760166

### CODE REVIEW

File: .github/workflows/build-docker-image-and-binaries.yaml [hunk 1]
```diff
@@ -48,27 +50,44 @@ jobs:
...
+          OS="darwin"
+          ARCH="${{ matrix.arch }}" 
+          if [ "$ARCH" == "arm64" ]
           then
-              OS="linux"
-              TOFND_PATH="/home/runner/work/tofnd/tofnd/target/release"
+            export HOMEBREW_NO_INSTALL_CLEANUP=TRUE
+            brew uninstall --ignore-dependencies gmp
+            rustup target add aarch64-apple-darwin
+            ARM_DEPENDENCY=$(brew fetch --force --bottle-tag=arm64_big_sur gmp | grep Downloaded | awk '{print $3}')
```
Comment (milapsheth) [inline .github/workflows/build-docker-image-and-binaries.yaml:65]: I prefer to keep the workflow simple and move the case specific logic into a script rather have separate case dependent steps.

File: .github/workflows/build-docker-image-and-binaries.yaml [hunk 2]
```diff
@@ -170,4 +189,4 @@ jobs:
         run: cosign sign --oidc-issuer https://token.actions.githubusercontent.com ${TAGS}
         env:
           TAGS: axelarnet/tofnd:${{ github.event.inputs.tag }}
-          COSIGN_EXPERIMENTAL: 1
+          COSIGN_EXPERIMENTAL: 1
```
Comment (milapsheth) [inline .github/workflows/build-docker-image-and-binaries.yaml:192]: Keep newlines at the end of files?

File: .github/workflows/build-docker-image-and-binaries.yaml [hunk 3]
```diff
@@ -7,14 +7,16 @@ on:
...
 
+jobs:
   release-binaries:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ubuntu-18.04, macos-latest]
         arch: [amd64, arm64]
+        exclude:
+          - {os: "ubuntu-18.04", arch: "arm64"}
```
[the remark this answers is not in the document]
Comment (kalidax) [reply .github/workflows/build-docker-image-and-binaries.yaml:19]: Yes
For cross compiling we need to install gmp for ARM64.
We need to add ARM repos to  /etc/apt/sources.list, and make apt install X:arm64 (where X is the package to install)
see https://askubuntu.com/questions/1169165/installing-libraries-for-arm64-on-ubuntu-bionic-x64 
However we are not allowed to add repos in sources.list file using github actions.

File: .github/workflows/build-docker-image-and-binaries.yaml (hunk 1 above)
Comment (kalidax) [reply .github/workflows/build-docker-image-and-binaries.yaml:65]: sounds good. I will encapsulate the commands

Comment (kalidux) [reply .github/workflows/build-docker-image-and-binaries.yaml]: I added a script to encapsulate the complexity (install ARM64 dependency)

**Review by talalashraf — APPROVED** [review]
looks good. can we add something in README to compile for whatever platform we are missing. Can be a separate PR.

**Comment by kalidax** [comment]
> looks good. can we add something in README to compile for whatever platform we are missing. Can be a separate PR.

Ok, I will add it.

**Review by sdaveas — APPROVED** [review]

**Pull request closed without merging**